### PR TITLE
[AAE-16369] use layout-bp mixin where applicable

### PR DESCRIPTION
**Pull request by wojd0**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [x] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

https://alfresco.atlassian.net/browse/AAE-15828

**What is the new behaviour?**

Replaced `@media` rules where it was possible with `@mixin layout-bp` where it was possible to encourage the use of global variables for responsive breakpoints.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by eromano** [comment]
@wojd0 if you search for  "media screen and " you will find many more occurrences. Let's use it for
max-width: 959px and 960 px for small devices
max-width:599 and 600px for x-small

File: lib/core/src/lib/layout/components/header/header.component.scss [hunk 1]
```diff
@@ -1,3 +1,5 @@
+@import '../../../styles/_flex.scss';
```
Comment (DenysVuika) [inline lib/core/src/lib/layout/components/header/header.component.scss:1]: underscore in the import

Comment (wojd0) [reply lib/core/src/lib/layout/components/header/header.component.scss]: fixed

**Comment by wojd0** [comment]
@eromano should we also use `layout-bp` in the demo app?

**Review by eromano — APPROVED** [review]

**Comment by DenysVuika** [comment]
@wojd0 can you investigate please on how to import via the project name, instead of this scary `@import '../../../../core/src/lib/styles/flex';` things

**Comment by wojd0** [comment]
Absolutely right @DenysVuika. I'll apply this solution.

**Comment by wojd0** [comment]
After some configuartion additions I managed to shorten the import paths for scss files. 
Now you can import anything from the core/lib path in scss like it is in the root dir for example `@import 'styles/flex` points to `/core/src/lib/styles/flex` file.

**Review by DenysVuika — APPROVED** [review]

**Pull request merged**